### DISCUSSION TITLE
Is it possible to send additional arguments to createUser when a new user is created?

### DISCUSSION THREAD

**Discussion by dgrahamlong** [nextauthjs/next-auth#4650] [Ideas]
### Question 💬

Hi, I was wondering if it was possible to send additional arguments when creating a new user. 

I'm trying to create an invite-only/waitlist signup process and want to use a verification token to ensure the said user can signup. 

I can use the ```signIn``` configuration callback to ensure the user's email is approved for signing up however, this can lead to a bad UX for users who decide to sign up with another email or use a provider like Facebook which sometimes doesn't return an email.

It also slows the sign-in process down because I have to check for approved emails regardless of it being a new user or not. ```additional parameters``` doesn't seem to provide what I'm suggesting. Maybe I'm missing something??

I was thinking if I could send an additional token, sort of like email verification, then I could allow approved users to sign up however they want, whether they're using a different email or email-less provider, etc.

### How to reproduce ☕️

I'm thinking of something like this for the signIn callback
```
callbacks: {
    async signIn(args) {
      if(args.newUser) {
          if(await checkIfUserCanSignUp(args.additionalParams.customToken)){
                return true;
          }
          return false;
      }
      return true;
    }
}

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by dgrahamlong** [comment]
#1069 seems to be in the same ballpark of what I'm trying to accomplish

**Comment by daniellin215** [comment]
Bumping as I and many others have asked for this many times both in discussions and issues but seems like the team has moved on

**Comment by dgrahamlong** [comment]
Created a pull request for this #4747
